TTS/tts/models/base_tts.py

```diff
@@ -355,12 +356,22 @@ def get_data_loader(
             ), "speaker_weighted_sampler is not supported with DistributedSampler"
 
             if sampler is None:
+                weights = None
                 if getattr(config, "use_language_weighted_sampler", False):
-                    print(" > Using Language weighted sampler")
-                    sampler = get_language_weighted_sampler(dataset.items)
-                elif getattr(config, "use_speaker_weighted_sampler", False):
-                    print(" > Using Language weighted sampler")
-                    sampler = get_speaker_weighted_sampler(dataset.items)
+                    alpha = getattr(config, "language_weighted_sampler_alpha", 1.0)
+                    print(" > Using Language weighted sampler with alpha:", alpha)
+                    weights = get_language_balancer_weights(dataset.items) * alpha
+
+                if getattr(config, "use_speaker_weighted_sampler", False):
+                    alpha = getattr(config, "speaker_weighted_sampler_alpha", 1.0)
+                    print(" > Using Speaker weighted sampler with alpha:", alpha)
+                    if weights is not None:
+                        weights += get_speaker_balancer_weights(dataset.items) * alpha
+                    else:
+                        weights = get_speaker_balancer_weights(dataset.items) * alpha
```

Comment:
I think gathering all the samplers and weighters in TTS/tts/datasets/samplers.py and using a factory pattern to pull the right set of samplers would be a more convenient solution.

Reply:
Agreed, but we will change all of it for multi GPU support so I just kept it in the way that it was. Do you want this change or we can do it when we will implement the multi GPU support? (I guess that it will change a lot)

Reply:
get_data_loader got too long. Somehow we need to split it so that we can override different parts in different models.  

Maybe 

```BaseTTS.get_sampler()``` ?

Reply:
Done, I added get_sampler in BaseTTS.